farmerbot/internal/farmerbot.go

```diff
@@ -0,0 +1,425 @@
+			return nil, fmt.Errorf("node %d is excluded, cannot add it", nodeID)
+		}
+
+		neverShutDown := slices.Contains(f.config.NeverShutDownNodes, nodeID)
+		node, err := getNode(ctx, subConn, f.rmbNodeClient, nodeID, neverShutDown, false, f.farm.DedicatedFarm, on)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to include node with id %d", nodeID)
+		}
+
+		f.state.addNode(node)
+		return nil, nil
+	})
+
+	powerRouter.WithHandler("poweroff", func(ctx context.Context, payload []byte) (interface{}, error) {
+		err := authorize(ctx, farmerTwinID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := f.validateAccountEnoughBalance(subConn); err != nil {
+			return nil, fmt.Errorf("failed to validate account balance: %w", err)
+		}
+
+		var nodeID uint32
+		if err := json.Unmarshal(payload, &nodeID); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		if err := f.powerOff(subConn, nodeID); err != nil {
+			return nil, fmt.Errorf("failed to power off node %d: %w", nodeID, err)
+		}
+
+		// Exclude node from farmerbot management
+		// (It is not allowed if we tried to power on a node the farmer decided to power off)
+		// the farmer should include it again if he wants to the bot to manage it
+		f.state.deleteNode(nodeID)
+		return nil, nil
+	})
+
+	powerRouter.WithHandler("poweron", func(ctx context.Context, payload []byte) (interface{}, error) {
+		err := authorize(ctx, farmerTwinID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := f.validateAccountEnoughBalance(subConn); err != nil {
+			return nil, fmt.Errorf("failed to validate account balance: %w", err)
+		}
+
+		var nodeID uint32
+		if err := json.Unmarshal(payload, &nodeID); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		if err := f.powerOn(subConn, nodeID); err != nil {
+			return nil, fmt.Errorf("failed to power on node %d: %w", nodeID, err)
+		}
+
+		// Exclude node from farmerbot management
+		// (It is not allowed if we tried to power off a node the farmer decided to power on)
+		// the farmer should include it again if he wants to the bot to manage it
+		f.state.deleteNode(nodeID)
+		return nil, nil
+	})
+
+	_, err = peer.NewPeer(
+		ctx,
+		f.mnemonicOrSeed,
+		subConn,
+		router.Serve,
+		peer.WithRelay(relayURLS[f.network]),
+		peer.WithSession(fmt.Sprintf("farmerbot-%d", f.farm.ID)),
+	)
+
+	if err != nil {
+		return errors.Wrap(err, "failed to create farmerbot direct peer")
+	}
+
+	return nil
+}
+
+func (f *FarmerBot) iterateOnNodes(ctx context.Context) error {
+	roundStart := time.Now()
+	var wakeUpCalls uint8
+
+	subConn, err := f.substrateManager.Substrate()
+	if err != nil {
+		return err
+	}
+	defer subConn.Close()
+
+	log.Debug().Msg("Fetch nodes")
+	farmNodes, err := subConn.GetNodes(uint32(f.state.farm.ID))
+	if err != nil {
+		return err
+	}
+
+	// remove nodes that doesn't exist anymore in the farm
```

Comment:
don't exist on the farm or the farmer didn't include them in the range specified?

Reply:
no, If it was included and for any reason the node doesn't exist anymore in the farm.